### PR TITLE
#160577570 Enable registered user to login 

### PR DESCRIPTION
**Pull request by C3real-kill3r**
**What does this PR do?**

creates a login feature that enables registered users to gain access into the application.

This pull request creates the following functionalities
* creates login form
* creates login reducer.
* creates action creators to dispatch actions.
* stores token and user profile in local storage on successful login

**Screenshots**

* Login Page
screenshot 2018-11-09 at 00 47 57

* Login with wrong credentials
screenshot 2018-11-09 at 00 48 37

Successful login should redirect to the homepage

**How should this be manually tested?**

- clone this repository.
- branch out to ft-login-#160577570
- install node and yarn
- run `yarn install`
-  run `yarn start`
- navigate to `/user/login/`

**What are the relevant pivotal tracker stories?**

[#160577570](https://www.pivotaltracker.com/story/show/160577570)

### CODE REVIEW

**Review by charisschomba — COMMENTED** [review]
Looks good @C3real-kill3r  fix hound flags

**Review by YAHYA-H — APPROVED** [review]
GOOD WORK! 👍 , you can now work on those two Hound issues

**Review by bevkololi — APPROVED** [review]
Good job @C3real-kill3r . Ensure to integrate with signup as requested by PO

File: src/containers/LoginForm/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import * as M from "materialize-css";
```
Comment (paulrimiru) [inline src/containers/LoginForm/index.js]: Change this into something more descriptive

File: src/containers/LoginForm/state/actions.test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  describe('loginin failure action creator', () => {
+    it('should create an action to dispatch errors to the user', () => {
+      const error = err;
+      const expectedAction = {
+        type: LOGIN_FAILURE,
+        payload: err,
+      };
+      expect(loginFailure(error)).toEqual(expectedAction);
+    });
+  });
```
Comment (paulrimiru) [inline src/containers/LoginForm/state/actions.test.js:71]: Add new line

File: src/containers/LoginForm/state/actions.test.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  describe('loginin success action creator', () => {
+    it('should create an action to dispatch success response.', () => {
+      const data = resData;
+      const expectedAction = {
+        type: LOGIN_SUCCESS,
+        data,
+      };
+      expect(loginSuccess(data)).toEqual(expectedAction);
+    });
+  });
```
Comment (paulrimiru) [inline src/containers/LoginForm/state/actions.test.js:82]: Add a new line

File: src/containers/LoginForm/state/reducer.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { LOGIN_SUCCESS, LOGIN_FAILURE, LOGIN_START } from './types';
+
+const initialState = {
+  data: {},
+  errors: {},
+  success: false,
+  isLogingIn: false,
+};
```
Comment (paulrimiru) [inline src/containers/LoginForm/state/reducer.js:8]: Add new line

File: src/pages/SocialLogin/index.js [hunk 5]
```diff
@@ -23,7 +23,7 @@ class SocialLogin extends Component {
   }
 
   /* istanbul ignore next */
-  responseFacebook(response) {
+  responseFacebook=(response) => {
```
Comment (paulrimiru) [inline src/pages/SocialLogin/index.js]: Pad the `=` sign with a space

**Review by paulrimiru — CHANGES_REQUESTED** [review]
Fix these issues and we can merge

File: src/containers/LoginForm/state/reducer.js (hunk 4 above)
Comment (C3real-kill3r) [reply src/containers/LoginForm/state/reducer.js:8]: done

File: src/pages/SocialLogin/index.js (hunk 5 above)
Comment (C3real-kill3r) [reply src/pages/SocialLogin/index.js]: noted and fixed

File: src/containers/LoginForm/state/actions.test.js (hunk 3 above)
Comment (C3real-kill3r) [reply src/containers/LoginForm/state/actions.test.js:82]: done

File: src/containers/LoginForm/state/actions.test.js (hunk 2 above)
Comment (C3real-kill3r) [reply src/containers/LoginForm/state/actions.test.js:71]: done

File: src/containers/LoginForm/index.js (hunk 1 above)
Comment (C3real-kill3r) [reply src/containers/LoginForm/index.js]: done

**Review by paulrimiru — APPROVED** [review]

**Pull request merged**